File: spring/errorcontroller-throw-exception-example/src/main/java/com/example/errorcontrollerthrowexceptionexample/MyController.java
package com.example.errorcontrollerthrowexceptionexample;

import java.util.Map;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("")
public class MyController {

    @GetMapping("")
    Map<String, String> index() {
        throw new RuntimeException();
    }
}

File: spring/thymeleaf/date-input/src/main/kotlin/com/example/date_input/MyController.kt
package com.example.date_input

import org.springframework.format.annotation.DateTimeFormat
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ModelAttribute
import java.time.LocalDate

@Controller
class MyController {

    @ModelAttribute("dto")
    fun dto(): Dto {
        return Dto()
    }

    @GetMapping
    fun index(model: Model): String {
        return "index"
    }
}

data class Dto(
//    @DateTimeFormat(pattern = "yyyy-MM-dd")
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    val date: LocalDate? = LocalDate.now(),
)
